sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/BroadcastExchangeExec.scala

```diff
@@ -82,6 +82,8 @@ case class BroadcastExchangeExec(
     "buildTime" -> SQLMetrics.createTimingMetric(sparkContext, "time to build"),
     "broadcastTime" -> SQLMetrics.createTimingMetric(sparkContext, "time to broadcast"))
 
+  private var knownRowCount: Option[BigInt] = None
```

Comment:
can we be consistent and use `metrics` to hold rowCount? You can follow how we track `dataSize`.

Reply:
Sure will do the update.